src/pineko/fonll.py

```diff
@@ -45,6 +45,20 @@ class InconsistentInputsError(Exception):
     """Raised if the inputs are not consistent with FONLL."""
 
 
+def _json_theory_read(theory_info):
+    """Read the theory info from a pineappl grid.
+
+    Theory information within pineappl grids might not be saved in a consistent
+    json-compatible way. This functions corrects some found problems.
+    """
+    try:
+        ret = json.loads(theory_info)
+    except json.decoder.JSONDecodeError:
+        # Some grids have the theories saved with incompatible formats
+        ret = json.loads(theory_info.replace("'", '"').replace("True", "true"))
+    return ret
```

Comment:
```suggestion
        return json.loads(theory_info)
    except json.decoder.JSONDecodeError:
        # Some grids have the theories saved with incompatible formats
        return json.loads(theory_info.replace("'", '"').replace("True", "true"))
```

Reply:
why?

(in this cases I prefer to leave the return outside so that it is clear that they should be the same object, but I don't know whether there's a reason to have it separated)

Reply:
Early return: as soon as you finish, return :)

It just makes the code more readable (slightly, in this case).

The output of a function is only one, though conditional. So, whatever you return is the same object (i.e. `ret = ...; return ret` is as good as `return ...` at telling that they are the same).